ibc-primitives/src/types/duration.rs

```diff
@@ -0,0 +1,51 @@
+   use core::time::Duration;
+   use ibc_proto::google::protobuf::Duration as RawDuration;
+   use crate::prelude::*;
+
+   /// Converts a core::time::Duration into a protobuf Duration
+   pub fn duration_to_proto(d: Duration) -> Option<RawDuration> {
+       let seconds = i64::try_from(d.as_secs()).ok()?;
+       let nanos = i32::try_from(d.subsec_nanos()).ok()?;
+       Some(RawDuration { seconds, nanos })
+   }
```

Comment:
Not really sure if you understood my original issue because I need to have `RawDuration: From<Duration>`.